src/pkg/cli/client/byoc/aws/byoc.go

```diff
@@ -725,20 +729,23 @@ func (b *ByocAws) QueryLogs(ctx context.Context, req *defangv1.TailRequest) (cli
 		if req.Until.IsValid() {
 			end = req.Until.AsTime()
 		}
-		tailStream, err = ecs.QueryAndTailLogGroups(ctx, start, end, b.getLogGroupInputs(etag, req.Project, service, req.Pattern, logType)...)
+		tailStream, err = ecs.QueryAndTailLogGroups(ctx, start, end, b.getLogGroupInputs(etag, req.Project, service, req.Pattern, logs.LogType(req.LogType))...)
 		taskArn = b.cdTaskArn
+		stopWhenCDTaskDone = false
 	}
 	if err != nil {
 		return nil, AnnotateAwsError(err)
 	}
 	if taskArn != nil {
+		// When we have a CD task, override the context so we can cancel the tailing when the CD task is done
 		var cancel context.CancelCauseFunc
 		ctx, cancel = context.WithCancelCause(ctx)
 		go func() {
 			if err := ecs.WaitForTask(ctx, taskArn, 2*time.Second); err != nil {
-				if stopWhenCDTaskDone || errors.As(err, &ecs.TaskFailure{}) {
+				isTaskFailure := !errors.Is(err, io.EOF)
```

Comment:
I'm confused to read that the name `isTaskFailure` now means `notEOF`. It seems like `errors.As(err, &ecs.TaskFailure{})` is stronger assertion than `!errors.Is(err, io.EOF)`. So, why was this change made?